docs: add better docs for SSR support (#1490)
ionic-team/stencil-site: docs/framework-integration/react.md

export default App;
```

### Enable Server Side Rendering (SSR)

If your React framework supports server side rendering, e.g. [Next.js](https://nextjs.org/) your Stencil components will get automatically server side rendered, if set up correctly. In order to enable this:

1. Add a `dist-hydrate-script` output target to your `stencil.config.ts` if not already existing, e.g.:
    ```ts title="stencil.config.ts"
    import { Config } from '@stencil/core';

    export const config: Config = {
      outputTargets: [
        {
          type: 'dist-hydrate-script',
          dir: './hydrate',
        },
        // ...
      ]
    };
    ```

2. Create an export for the compiled files within the `/hydrate` directory, e.g.

>     ```json

Reviewer: ```suggestion
    ```json title="package.json"
```

    {
      "name": "component-library",
      ...
      "exports": {
        ...
        "./hydrate": {
          "types": "./hydrate/index.d.ts",
          "import": "./hydrate/index.js",
          "require": "./hydrate/index.cjs.js",
          "default": "./hydrate/index.js"
        },
        ...
      },
      ...
    }
    ```

3. Set the `hydrateModule` in your React output target configuration, e.g.

>     ```ts

Reviewer: ```suggestion
    ```ts title="stencil.config.ts"
```

    import { Config } from '@stencil/core';
    import { reactOutputTarget } from '@stencil/react-output-target';

    export const config: Config = {
      outputTargets: [
        reactOutputTarget({
          outDir: '../react-library/lib/components/stencil-generated/',
          hydrateModule: 'component-library/hydrate'
        }),
        // ...
      ]
    };
    ```

That's it! Your Next.js application should now render a Declarative Shadow DOM on the server side which will get automatically hydrated once the React runtime initiates.

:::cautions

> A Declarative Shadow DOM contains next to the HTML structure also all the CSS defined for the component. If you server side render a lot of small components that come with large amounts of CSS, it will drastically increase the initial page load time as the documented loaded by the browser increases in size. Make sure to keep the initial document size reasonable and aligned with your performance goals, server side rendering only the critical components needed to render the viewport and load the rest later on.

Reviewer: I'm a bit unclear about the first part of the first sentence: "A Declarative Shadow DOM contains next to the HTML structure"

Couple other tweaks outside of that:

```suggestion
A Declarative Shadow DOM contains next to the HTML structure also all the CSS defined for the component. If you server side render a lot of small components that come with large amounts of CSS, it will drastically increase the initial page load time as the document loaded by the browser increases in size. Make sure to keep the initial document size reasonable and aligned with your performance goals. It is recommended to server side rendering only the critical components needed to render the viewport and load the rest later on.
```
